### PR TITLE
Add fix for skipping fill bytes from pdf.js issue 5331

### PR DESCRIPTION
**Pull request by tim-king**
Some JPEG files end up giving the error "unknown jpeg marker ffff". If found this fix in Mozilla's pdf.js - it needs to be added to jpeg-js.

### CODE REVIEW

File: lib/decoder.js [hunk 1]
```diff
@@ -747,6 +747,13 @@ var JpegImage = (function jpegImage() {
               successiveApproximation >> 4, successiveApproximation & 15);
             offset += processed;
             break;
+
+            case 0xFFFF: // Fill bytes
```
Comment (patrickhulce) [inline lib/decoder.js]: nit: this seems to be misaligned with the other case statements?

Comment (tim-king) [inline lib/decoder.js]: nit: feel free to change
Example bad file attached - this was one from a CCTV system that tripped me up.
mpt dgos_camera3_mpt dgos_20170813152826_3955415

**Comment by jginternational** [comment]
Please merge this PR! It is needed in  [oliver-moran/jimp](/oliver-moran/jimp)

**Comment by tim-king** [comment]
Yes please it's a pain I just had to put in symlinks in my npm file 
structure to my workaround on three EC2 instances because it's not 
merged in.

JaviG wrote:

**Comment by paulirish** [comment]
@tim-king can you write the test for this provided asset? and fix the indentation?

**Comment by tim-king** [comment]
How am I supposed to create another image with the identical 
rawImageData that the other tests check? Tried exporting rgb from Gimp 
and that doesn't work.

Paul Irish wrote:

**Comment by patrickhulce** [comment]
@tim-king you can skip the RGBA file check if the problems was that it errored on decode before

those are essentially golden files that assert the output of jpeg-js doesn't change between versions, they are just the buffer output of rawImageData written to disk

**Comment by carterbancroft** [comment]
Thank you for this fix! Would it be possible to get a release on this now that it's merged into master? As mentioned above this change would fix this issue with jimp https://github.com/oliver-moran/jimp/issues/177.

**Comment by patrickhulce** [comment]
@carterbancroft just published 0.3.4 👍

**Comment by carterbancroft** [comment]
@patrickhulce thanks for the fast response/release! Most appreciated.

**Pull request merged**